cmds/webboot/utils.go

```diff
@@ -7,8 +7,23 @@ import (
 	"net/http"
 	"net/url"
 	"os"
+	"strings"
 )
 
+// WriteCounter counts the number of bytes written to it. It implements to the io.Writer
+type WriteCounter struct {
+	Total uint64
+}
+
+func (wc *WriteCounter) Write(p []byte) (int, error) {
+	n := len(p)
+	wc.Total += uint64(n)
+	// print how many bytes have been writen to the file
+	fmt.Printf("\r%s", strings.Repeat(" ", 40))
+	fmt.Printf("\rDownloading... %v bytes complete", wc.Total)
```

Comment:
How often will this message print?

Reply:
It depends on how offten io.Copy() read a segment of data from the online file. I think it may involves  the speed of internet. In the test the message update many times per second.

Reply:
since we use /r here, the new message just overwrite the old message instead of create a new line. if that is what your worry about

Reply:
Ok - I'll approve this but would you mind also including a quick screengrab of what it looks like here for future reference?

Reply:
ok will do